bitbots_localization:
  ros__parameters:
    misc:
      init_mode: 0
      percentage_best_particles: 100
      max_motion_linear: 0.5
      max_motion_angular: 3.14
    ros:
      line_pointcloud_topic: 'line_mask_relative_pc'
      goal_topic: 'goals_simulated'
      particle_publishing_topic: 'pose_particles'
      debug_visualization: true
    particle_filter:
      particle_number: 300
      rate: 10
      resampling_interval: 2
      diffusion:
        x_std_dev: 0.8
        y_std_dev: 0.8
        t_std_dev: 0.9
        multiplier: 0.001
        starting_multiplier: 0.05
        starting_steps_with_higher_diffusion: 40
      drift:
        distance_to_direction: 1.0
        rotation_to_direction: 0.0
        distance_to_distance: 0.1
        rotation_to_distance: 0.2
        distance_to_rotation: 0.2
        rotation_to_rotation: 1.5
        max_rotation: 0.45
        max_translation: 0.09
      weighting:
        min_weight: 0.01
        particle_reset_weight: 0.01
        out_of_field_weight_decrease: 0.01
        out_of_field_range: 0.5
      scoring:
        lines:
          factor: 1.0
          out_of_field_score: 0.0
        goal:
          factor: 0.0
          out_of_field_score: 0.0
      confidences:
        line_element: 0.01
        goal_element: 0.0
